activation/handler.go

```diff
@@ -394,25 +398,35 @@ func (h *Handler) storeAtx(ctx context.Context, atx *types.ActivationTx) (*mwire
 		}
 
 		// do ID check to be absolutely sure.
-		if prev != nil && prev.ID() != atx.ID() {
+		if err == nil && prev != atx.ID() {
 			if _, ok := h.signers[atx.SmesherID]; ok {
 				// if we land here we tried to publish 2 ATXs in the same epoch
 				// don't punish ourselves but fail validation and thereby the handling of the incoming ATX
 				return fmt.Errorf("%s already published an ATX in epoch %d", atx.SmesherID.ShortString(),
 					atx.PublishEpoch,
 				)
 			}
+			prevSignature, err := atxSignature(ctx, tx, prev)
+			if err != nil {
+				return err
```

Comment:
I think we should add some context here:

```suggestion
				return fmt.Errorf("extracting signature for malfeasance proof: %w")
```

or something similar so it is clear the error during handling the ATX isn't caused by the ATX itself but by failing to create a proof for an apparently malfeasant ATX.